sys/include/vfs.h

```diff
@@ -229,6 +230,27 @@ extern "C" {
  */
 #define VFS_ANY_FD (-1)
 
+/**
+ * @brief Define an automatic mountpoint
+ *
+ * @param[in] type  file system type
+ *                  Can by littlefs, littlefs2, spiffs or fatfs
+ * @param[in] mtd   MTD device to use
+ * @param[in] path  Mount path
+ */
+#define VFS_AUTO_MOUNT(type, mtd, path)     \
+    static type ## _desc_t fs_desc = {      \
+        .dev = (mtd_dev_t *)(mtd),          \
```

Comment:
Is the casting necessary?

Reply:
Yes because `mtd` could be `mtd_native_dev_t`, `mtd_spi_nor_t`, `mtd_at24cxxx_t` or `mtd_sdcard_t`

Reply:
But can't we enforce that they pass the pointer to the `mtd_dev_t` field always?

Reply:
turns out `native` is the only odd one out here - we can fix this and just always use `&mtd.base`